kanidmd/src/lib/valueset.rs

```diff
@@ -1,112 +1,1297 @@
+        match &mut self.inner {
+            I::Utf8(set) => {
+                set.extend(iter.filter_map(|v| v.to_utf8()));
+            }
+            I::Iutf8(set) => {
+                set.extend(iter.filter_map(|v| v.to_iutf8()));
+            }
+            I::Iname(set) => {
+                set.extend(iter.filter_map(|v| v.to_iname()));
+            }
+            I::Uuid(set) => {
+                set.extend(iter.filter_map(|v| v.to_uuid().copied()));
+            }
+            I::Bool(set) => {
+                iter.filter_map(|v| v.to_bool()).for_each(|i| {
+                    set.insert(i);
+                });
+            }
+            I::Syntax(set) => {
+                iter.filter_map(|v| v.to_syntaxtype().cloned())
+                    .for_each(|i| {
+                        set.insert(i);
+                    });
+            }
+            I::Index(set) => {
+                iter.filter_map(|v| v.to_indextype().cloned())
+                    .for_each(|i| {
+                        set.insert(i);
+                    });
+            }
+            I::Refer(set) => {
+                set.extend(iter.filter_map(|v| v.to_ref_uuid().copied()));
+            }
+            I::JsonFilt(set) => {
+                iter.filter_map(|v| v.to_jsonfilt()).for_each(|i| {
+                    set.insert(i);
+                });
+            }
+            I::Cred(map) => {
+                map.extend(iter.filter_map(|v| v.to_cred()));
+            }
+            I::SshKey(map) => {
+                map.extend(iter.filter_map(|v| v.to_sshkey()));
+            }
+            I::SecretValue(set) => {
+                iter.filter_map(|v| v.get_secret_str().map(str::to_string))
+                    .for_each(|i| {
+                        set.insert(i);
+                    });
+            }
+            I::Spn(set) => {
+                set.extend(iter.filter_map(|v| v.to_spn()));
+            }
+            I::Uint32(set) => {
+                iter.filter_map(|v| v.to_uint32()).for_each(|i| {
+                    set.insert(i);
+                });
+            }
+            I::Cid(set) => {
+                iter.filter_map(|v| v.to_cid()).for_each(|i| {
+                    set.insert(i);
+                });
+            }
+            I::Nsuniqueid(set) => {
+                set.extend(iter.filter_map(|v| v.to_nsuniqueid()));
+            }
+            I::DateTime(set) => {
+                iter.filter_map(|v| v.to_datetime()).for_each(|i| {
+                    set.insert(i);
+                });
+            }
+            I::EmailAddress(set) => {
+                set.extend(iter.filter_map(|v| v.to_emailaddress()));
+            }
+            I::Url(set) => {
+                iter.filter_map(|v| v.to_url().cloned()).for_each(|i| {
+                    set.insert(i);
+                });
+            }
+        }
+    }
+
+    pub(crate) fn clear(&mut self) {
+        match &mut self.inner {
```

Comment:
If we change the inner type of the set the match would become invalid then because of the conflicting type of the set, so I'd rather duplicate and repeat than have to go through a more painful refactor later if we change up the inner set type.